src/codegen/compile.ml

```diff
@@ -10627,7 +10627,9 @@ and compile_prim_invocation (env : E.t) ae p es at =
          StackRep.of_arity return_arity,
 
          code1 ^^
-         compile_unboxed_zero ^^ (* A dummy closure *)
+         Type.(match as_obj par.note.Note.typ with
+         | Object, [] -> compile_unboxed_zero (* a dummy closure *)
```

Comment:
Perhaps push a dummy, zero-length closure, so that `ICCyclesPrim` doesn't need to do a null-check?